ports/rp2/tusb_config.h

```diff
@@ -25,10 +25,20 @@
 #ifndef MICROPY_INCLUDED_RP2_TUSB_CONFIG_H
 #define MICROPY_INCLUDED_RP2_TUSB_CONFIG_H
 
+#include "mpconfigboard.h"
+
 #define CFG_TUSB_RHPORT0_MODE   (OPT_MODE_DEVICE)
 
 #define CFG_TUD_CDC             (1)
 #define CFG_TUD_CDC_RX_BUFSIZE  (256)
 #define CFG_TUD_CDC_TX_BUFSIZE  (256)
 
+#if MICROPY_HW_USB_MSC
+// Board and hardware specific configuration
+#define CFG_TUD_MSC             (1)
+// This must match both the FLASH_SECTOR_SIZE and FATFS_MAX_SS
+// Use 4096 to avoid caching.
```

Comment:
What you mean by that? How do they have to match? Should they be exactly the same?

What caching do you refer to, here?

Reply:
FatFS block size (and CFG_TUD_MSC_BUFSIZE) must be set to the same size as the flash sector (4K), this is the easy way to implement this, otherwise we have load, modify, flush flash sectors to support smaller block sizes.

Reply:
So maybe that should actually be asserted here? Something like:
```c
#if FLASH_SECTOR_SIZE != FATFS_MAX_SS
#error FLASH_SECTOR_SIZE should be the same as FATFS_MAX_SS
#endif
#define CFG_TUD_MSC_BUFSIZE FLASH_SECTOR_SIZE
```

Reply:
Okay but will have to include the flash header to get that value.

Reply:
I think it's fair to do this is the code correctness depends on this.

Reply:
Actually MSC EP buffer size just needs to be the same size as the FS block size `MICROPY_FATFS_MAX_SS` so tinyusb doesn't sent partial read/writes (offset arg will not be used). And in mpconfigport.h I set `MICROPY_FATFS_MAX_SS` to `FLASH_SECTOR_SIZE` and added a comment. And I moved that check to msc_disk.c instead.